### PR TITLE
Feature: Select Winner

### PR DESCRIPTION
**Pull request by jerixmx**
### What this PR does (required):
- Enables selecting winning submission with radio buttons
- Allows selecting from multiple payment methods
- Charges card according to the prize amount
- Checks with multiple criteria before allowing winner selection:
-- contest deadline, contest ownership, availability of payment method
- Changes button based on whether logged in user is contest owner
- Fixes other people's submissions not showing up when contest owner views contest details page

### Screenshots / Videos (required):
https://user-images.githubusercontent.com/78225194/128404654-fc76ba89-5dfa-4b80-b696-ea924eddde41.mp4

### Any information needed to test this feature (required):
- Access another user's contest from the Discovery Page. The button should show Submit Design instead of Select.
- Access own contest and attempt to select a winner before the deadline.
- Access own contest where deadline has passed. Should be able to select a winner IF there is a payment method. 
- Try to use different cards when paying. Check with Stripe if the correct card and amount are used.
- Check in the database whether the correct submission is tagged as the winner.

### Any issues with the current functionality (optional):
- Look of the Payments page still needs work. Winning submission can be added.
- Since the contest details tab is currently showing all the files (instead of a featured photo per submission), multiple files are selected by the radio button because they belong to the same submission.
-- Submissions are also currently per user. Ideally, a customer may have multiple different sets of submissions.
- Contest is not tagged yet as "winner selected" or "payment due".
- No notifications yet when the deadline has passed or a winner had been selected.
- - A "receipt" page/component may be added.
- No link yet to Payments page when winner is already selected.

### CODE REVIEW

**Comment by jerixmx** [comment]
Apologies for the blinking video! Must be my graphics card driver

File: client/src/pages/ContestPayment/PaymentMethodSelection/PaymentMethodRadio.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const [value, setValue] = useState<string>(paymentMethods[0]?.id);
+
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value);
+  };
+
+  const handleSubmit = async () => {
+    console.log(contestId, value);
```
Comment (rajivtitus) [inline client/src/pages/ContestPayment/PaymentMethodSelection/PaymentMethodRadio.tsx]: Please remove console.log's throughout

File: client/src/pages/ContestPayment/PaymentMethodSelection/PaymentMethodRadio.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      paymentIntentStatus.status === 'succeeded'
+        ? updateSnackBarMessage(
+            `${paymentIntentStatus.currency} ${paymentIntentStatus.amount_received} has been charged from your account.`,
+          )
+        : updateSnackBarMessage(paymentIntentStatus.status);
+    } else {
+      updateSnackBarMessage(`Unable to charge`);
+    }
+  };
+  console.log(value);
```
Comment (rajivtitus) [inline client/src/pages/ContestPayment/PaymentMethodSelection/PaymentMethodRadio.tsx]: Please remove

**Review by rajivtitus — APPROVED** [review]
Good work with this! Just a minor comment, thanks!

**Pull request merged**